applications/ThermalDEMApplication/custom_elements/thermal_spheric_particle.cpp

```diff
@@ -148,15 +137,35 @@ namespace Kratos
     SetGenerationModel(generation_model);
     SetRealContactModel(real_contact_model);
 
+    // Set flags
+    mHasMotion = r_process_info[MOTION_OPTION];
+
+    if ((GetProperties().Has(THERMAL_EXPANSION_COEFFICIENT) && GetProperties()[THERMAL_EXPANSION_COEFFICIENT] != 0.0) ||
```

Comment:
you can declare the properties `r_properties` above and use the same one in all `GetProperties()` that you have there